src/model/linear-model/linear-regression.ts

```diff
@@ -0,0 +1,219 @@
+   * @param xData Tensor like of shape (n_samples, n_features), input feature
+   * @param yData Tensor like of shape (n_sample, ), input target values
+   * @param params batchSize: batch size: default to 32, maxIterTimes: max iteration times, default to 20000
+   * @returns classifier itself
+   */
+  public async fit(xData: Tensor | RecursiveArray<number>,
+    yData: Tensor | RecursiveArray<number>,
+    params: LinearRegerssionTrainParams = { batchSize: 32, epochs: -1 }): Promise<LinearRegression> {
+
+    const { x, y } = this.validateData(xData, yData);
+    const nFeature = x.shape[1];
+    const nData = x.shape[0];
+    const batchSize = nData > params.batchSize ? params.batchSize : nData;
+    const epochs = params.epochs > 0 ? params.epochs : null;
+    this.initLinearModel(nFeature, this.fitIntercept);
+    this.featureSize = nFeature;
+    await this.model.fit(x, y, {
+      batchSize,
+      epochs,
+      shuffle: true
+    });
+    return this;
+  }
+
+  public predict(xData: Tensor | RecursiveArray<number>): Tensor | Tensor[] {
+    const x = checkArray(xData, 'float32');
+    const predY = this.model.predict(x);
+    if (predY instanceof Tensor) return squeeze(predY);
+    return predY;
+  }
+
+  public initModelFromWeights(inputShape: number, useBias: boolean, weights: (Float32Array | Int32Array | Uint8Array)[]): void {
+    const weightsTensors = [];
+    for (const w of weights) {
+      weightsTensors.push(tensor(w));
+    }
+    this.initLinearModel(inputShape, useBias, weightsTensors);
+  }
+
+  /**
+   * Get linear regression coefficients
+   * @returns Linear regression coefficients with structure
+   * {'coefficient': Tensor, 'intercept': Tensor}
+   */
+  public getCoef(): { 'coefficients': Tensor, 'intercept': Tensor} {
+    return {
+      'coefficients': squeeze(this.model.getWeights()[0]),
+      'intercept': this.fitIntercept ? this.model.getWeights()[1] : tensor(0)
+    };
+  }
+
+  public async getModelWeightsArray(): Promise<RecursiveArray<number>> {
+    const weights = [];
+    for (const w of this.model.getWeights()) {
+      weights.push(await w.array());
+    }
+    return weights;
+  }
+
+  /**
+   * Load model paramters from json string object
+   * @param modelJson model json saved as string object
+   * @returns model itself
+   */
+  public async fromJson(modelJson: string): Promise<LinearRegression> {
+    const modelParams = JSON.parse(modelJson);
+    if (modelParams.name !== 'LinearRegression'){
+      throw new RangeError(`${modelParams.name} is not Linear Regression`);
+    }
+    const { fitIntercept, optimizerType, optimizerProps,
+      modelWeights, featureSize } = modelParams;
+    this.fitIntercept = (fitIntercept as boolean);
+    this.featureSize = featureSize;
+    if (optimizerType as OptimizerType && optimizerProps as OptimizerProps) {
```

Comment:
If you want to make a type judgment, you need to use the [`is` operator](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates)